satpy/readers/iasi_l2.py

```diff
@@ -175,3 +175,33 @@ def _form_datetimes(days, msecs):
         all_datetimes.append(scanline_datetimes)
 
     return np.array(all_datetimes, dtype=np.float64)
+
+
+class IASIL2CDRNC(NetCDF4FsspecFileHandler):
+    """Reader for IASI L2 CDR in NetCDF format.
+
+    Reader for IASI All Sky Temperature and Humidity Profiles - Climate
+    Data Record Release 1.1 - Metop-A and -B. Data and documentation are
+    available from http://doi.org/10.15770/EUM_SEC_CLM_0063. Data are
+    also available from the EUMETSAT Data Store under ID EO:EUM:DAT:0576.
+    """
+
+    def get_dataset(self, data_id, ds_info):
+        """Obtain dataset."""
+        ds = self[data_id["name"]]
+        if "scan_lines" in ds.dims:
+            ds = ds.rename(scan_lines="y")
+        if "pixels" in ds.dims:
+            ds = ds.rename(pixels="x")
+        return ds
+
+    def available_datasets(self, configured_datasets=None):
+        """Get available datasets based on what's in the file.
+
+        Returns all datasets in the root group.
+        """
+        yield from super().available_datasets(configured_datasets)
+        for key in self.file_content:
+            if "/" in key:  # not a dataset
+                continue
+            yield (True, {"name": key, "file_type": "iasi_l2_cdr_nc"} | self[key].attrs)
```

Comment:
I guess this will for sure require us to drop Python 3.8 since this `|` is a 3.9 feature. Not that we weren't planning to do that, but this makes it a requirement.

Reply:
Is there any `resolution` information that could be added?

Reply:
Yes.  I copied the resolution from the other IASI reader.